app/model/runtime_manager.py

```
@@ -0,0 +1,319 @@
+        # Use a short canonical sample to warm model
+        sample = "warmup"
+        for i in range(warmup_queries):
+            try:
+                coro = new_runtime.predict(sample, timeout_ms)
+                if asyncio.iscoroutine(coro):
+                    await coro
+                else:
+                    # in case implementation is sync
+                    # run in thread executor to avoid blocking loop
+                    loop = asyncio.get_running_loop()
+                    await loop.run_in_executor(None, lambda: coro)
```

Ruff (0.13.1)

Function definition does not bind loop variable `coro`

(B023)